### PR TITLE
Add ability to pass a custom input

### PR DESCRIPTION
**Pull request by linpe**
Addresses https://github.com/Hacker0x01/react-datepicker/issues/529

Added a `customInput` prop which allows you to pass in your own component, allowing you to add custom functionality and styling.

### CODE REVIEW

**Comment by martijnrusschen** [comment]
Thanks!

**Comment by cowboy** [comment]
Now that this has been added, can you publish a new release? Thanks!!

**Comment by i4got10** [comment]
Please add ``customInput`` prop to  [docs](https://github.com/Hacker0x01/react-datepicker/blob/master/docs/datepicker.md)

**Comment by martijnrusschen** [comment]
I'll take a look tomorrow.

**Comment by mtrabelsi** [comment]
**REALLY** coverage is 90.93%? you must be kidding : when using this custom input there is a lot of bugs that doesn't work : 

1. onBlur
2. Placeholder 
3. and much more

**Comment by rafeememon** [comment]
@mtrabelsi, feel free to help out to fix those bugs

**Comment by aij** [comment]
@rafeememon I'm fairly sure those are differences of opinions rather than actual bugs. Though FWIW, I found it confusing at first as well...

For example, placeholder is currently set per the `placeholderText` prop. If we were to "fix" that so that placeholder from customInput is used instead, then the users who expect `placeholderText` to be honored would be displeased.

Would it make sense to render customInput as a component instead of cloning an element? That would give users full control over the rendering of custom inputs. (Though we may want to still support cloning for compatibility.)

**Comment by rafeememon** [comment]
@aij, these very well may be things that should be changed, but it was more commentary on the attitude of the comment, not the content =)

File: docs-site/src/examples/custom_input.jsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          <br />
+          {"<DatePicker"}<br />
+              {"customInput={<ExampleCustomInput />}"}<br />
+              {"selected={this.state.startDate}"}<br />
+              {"onChange={this.handleChange} />"}
+        </code>
+      </pre>
+      <div className="column">
+        <DatePicker
+            customInput={<ExampleCustomInput />}
```
Comment (prakashvgr) [inline docs-site/src/examples/custom_input.jsx]: C(CustomInput) must be caps customInput={<ExampleCustomInput />}

Comment (martijnrusschen) [reply docs-site/src/examples/custom_input.jsx]: Feel free to update the example and submit a PR

**Pull request merged**